prog/dns_zone/dns_zone_nexus.rb

```diff
@@ -12,6 +12,11 @@ class Prog::DnsZone::DnsZoneNexus < Prog::Base
       register_deadline(:wait, 5 * 60)
       hop_refresh_dns_servers
     end
+
+    if dns_zone.last_purged_at < Time.now - 60 * 60 * 1 # ~1 hour
+      hop_purge_tombstoned_records
+    end
+
```

Comment:
I think we can move this block to higher than `when_refresh_dns_servers_set?` check just to keep the query cost lower on the refresh state. If we clean the tombstoned records before, the refresh will have to filter out less rows.